### PR TITLE
First version of clang-format based on DTK

### PR DESCRIPTION
**Pull request by aprokop**
Moved clang-format from #1. I don't have a strong opinion on what the options should be. @dalg24  has, so Let the Games Begin.

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,21 @@
+BasedOnStyle:  LLVM
+---
+AlwaysBreakTemplateDeclarations: true
+BreakBeforeBraces: Custom
+BreakConstructorInitializersBeforeComma: true
+IncludeBlocks: Regroup
+IndentWidth: 4
+Language: Cpp
+SpacesInParentheses: true
```
Comment (Rombur) [inline .clang-format]: Personally, I don't like this one. I am in favor of removing it.

**Comment by dalg24** [comment]
> Moved clang-format from #1. I don't have a strong opinion on what the options should be. @dalg24 has, so Let the Games Begin.

I actually do not.

I am fine with the changes suggested by Bruno.

**Comment by dalg24** [comment]
I am tempted to say let's integrate it into the CI as part of this PR

**Comment by aprokop** [comment]
One thing I'm not sure about is `BreakBeforeBraces`. Do we want `Allman` there (i.e., always break)?

**Comment by Rombur** [comment]
>  Do we want Allman there (i.e., always break)?

Yes, we do :D

**Comment by aprokop** [comment]
> I am tempted to say let's integrate it into the CI as part of this PR

OK.

**Comment by aprokop** [comment]
Rebased to resolve conflicts with `master`.

File: cmake/CodeFormat.cmake [hunk 2]
```diff
@@ -0,0 +1,45 @@
+if(NOT CLANG_FORMAT_EXECUTABLE)
+    find_program(CLANG_FORMAT_EXECUTABLE
+        NAMES
+        clang-format-7
+        clang-format-mp-7
```
Comment (Rombur) [inline cmake/CodeFormat.cmake]: Should we use clang-format-8 instead

Comment (aprokop) [reply cmake/CodeFormat.cmake]: I could not find it regular repos in Ubuntu.

Comment (Rombur) [reply cmake/CodeFormat.cmake]: http://apt.llvm.org/

Comment (aprokop) [reply cmake/CodeFormat.cmake]: Then I don't care. But we would need to rebuild the container as it carries 7.

**Comment by dalg24** [comment]
I am hesitant to merge that logic into CMake.  I'd rather just have the script.

**Comment by aprokop** [comment]
> I am hesitant to merge that logic into CMake. I'd rather just have the script.

Share your thoughts.

**Comment by dalg24** [comment]
This is just another chance to get the configuration fail and another argument to know about and to set when really all we want is a way to check/to enforce the style which the script is already doing

**Comment by aprokop** [comment]
@dalg24 I see your point now. I see an easy way to do that in Jenkins. However, that also adds a manual step for someone to do when trying to run tests locally if you don't want to touch cmake.

**Comment by aprokop** [comment]
OK, this is ready. Not going to change to clang-format 8 as of right now.

File: docker/jenkins/build.sh [hunk 3]
```diff
@@ -32,4 +32,7 @@ make -j${NPROCS}
 
 ctest --no-compress-output -T Test
 
+cd ${ARBORX_DIR}
+CLANG_FORMAT_EXE="clang-format" ./scripts/check_format_cpp.sh
```
Comment (dalg24) [inline docker/jenkins/build.sh]: Actually what about running the format check 1st?

Comment (aprokop) [reply docker/jenkins/build.sh]: That would mean that you won't see the results of the tests if your formatting fails. Formatting is secondary.

**Pull request merged**